File: Flutter/App/lib/domain/product/use_case/get_products_use_case.dart
import 'package:injectable/injectable.dart';
import 'package:products_crud/domain/core/use_case/async_use_case.dart';
import 'package:products_crud/domain/product/model/product.dart';
import 'package:products_crud/domain/product/repository/product_repository.dart';

@Injectable()
class GetProductsUseCase extends AsyncUseCase<void, List<Product>> {
  GetProductsUseCase(this._repository);

  final ProductRepository _repository;

  @override
  Future<List<Product>> call(void _) {
    return _repository.getProducts();
  }
}

File: Android/App/app/src/main/java/com/efalco777/crud_products/domain/product/use_case/GetProductsUseCase.kt
package com.efalco777.crud_products.domain.product.use_case

import com.efalco777.crud_products.core.interactor.UseCase
import com.efalco777.crud_products.domain.product.model.Product
import com.efalco777.crud_products.domain.product.repository.ProductRepository

class GetProductsUseCase (
    private val api: ProductRepository
) : UseCase<List<Product>, Unit>() {

    override suspend fun run(params: Unit):  List<Product> {
        return api.getProductList()
    }
}